### PR TITLE
FMDN upstream part 5: locator tag sample, FMDN and DULT modules

### PR DESCRIPTION
**Pull request by kapi-no**
Depends on the following parts (already merged):

Crypto Part 1:
https://github.com/nrfconnect/sdk-nrf/pull/15181
 
NVM Storage Part 2, 3 and 4:
https://github.com/nrfconnect/sdk-nrf/pull/15243
https://github.com/nrfconnect/sdk-nrf/pull/15275
https://github.com/nrfconnect/sdk-nrf/pull/15295

Necessary adaptions during the upstream activity:
- adaptation to HWMv2
- migration to sysbuild

Work items:
- [x] [Optional] Thingy:53 platform fails to boot after migration to sysbuild. This issue will be resolved as part of the sysbuild migration PR: https://github.com/nrfconnect/sdk-nrf/pull/13153

### CODE REVIEW

**Comment by kapi-no** [comment]
> Thingy:53 platform fails to boot after migration to sysbuild. This can probably be resolved later on, as the general sysbuild alignment PR aligns the build process for Thingy:53: https://github.com/nrfconnect/sdk-nrf/pull/13153

Confirmed with the newest hash of the sysbuild migration PR (https://github.com/nrfconnect/sdk-nrf/pull/13153):

https://github.com/nrfconnect/sdk-nrf/commit/9ec27c6958003a40a6390d44123f17c338da31f1

that sysbuild works for Thingy:53 in the Locator Tag sample.

**Review by annwoj — COMMENTED** [review]
Just some small suggestions

File: include/dult.h [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+ *  CONFIG_DULT_BATTERY Kconfig is disabled.
+ *
+ *  If the CONFIG_DULT_BATTERY Kconfig is enabled, this function must be called after
+ *  registering the DULT user with @ref dult_user_register and before enabling DULT with
+ *  @ref dult_enable function to initialize the battery level. Subsequent calls to
+ *  update the battery level are allowed in the enabled mode.
+ *
+ *  To keep the battery level information accurate, the user should set the battery level
+ *  to the new value with the help of this API as soon  as the device battery level
+ *  mandates such an update.
```
[the remark this answers is not in the document]
Comment (kapi-no) [reply include/dult.h:121]: yes, changing

**Comment by kapi-no** [comment]
Addressed @annwoj review comments regarding the API documentation

**Review by annwoj — APPROVED** [review]

**Pull request merged**